composer/datasets/in_context_learning_evaluation.py

```diff
@@ -1101,7 +1107,7 @@ def collate_fn(self, data):
             'test_outputs': test_outputs,  # list of test outputs
             'languages': languages,  # list of languages
             'pass_at_k': self.pass_at_k,
-            'generation_length': self.max_seq_len - self.max_prompt_length,
+            'generation_length': min(self.max_answer_length, self.max_seq_len - self.max_prompt_length),
```

Comment:
if max_answer_length is longer than max_seq_len - max_prompt_len, this would set the generation_length to max_seq_len - max_prompt_len. Is that the intended behavior?

Reply:
yeah, I think otherwise we will exceed max_seq_len, which should be a hard cutoff. Tradeoff is people shouldn't set max_seq_len too low. I can log an error/warning message if the condition you mentioned holds.